okapi-core/src/main/java/org/folio/okapi/util/DepResolution.java

```diff
@@ -137,54 +136,56 @@ public static void getLatestProducts(int limit, List<ModuleDescriptor> mdl) {
    * @param md          module to check.
    * @return            true if interface requirements are met
    */
-  static boolean moduleDepProvided(
-      List<ModuleDescriptor> modules, Set<String> allProvided,
+  public static boolean moduleDepProvided(
+      Collection<ModuleDescriptor> modules, Set<String> allProvided,
       ModuleDescriptor md) {
 
     for (InterfaceDescriptor req : md.getRequiresOptionalList()) {
-      if (allProvided.contains(req.getId())) {
-        boolean found = false;
-        for (ModuleDescriptor md1 : modules) {
-          InterfaceDescriptor[] providesList = md1.getProvidesList();
-          for (InterfaceDescriptor prov : providesList) {
-            if (prov.isRegularHandler() && prov.getId().equals(req.getId())) {
-              found = true;
-            }
+      if (!allProvided.contains(req.getId())) {
+        continue;
+      }
+      boolean found = false;
+      for (ModuleDescriptor md1 : modules) {
+        InterfaceDescriptor[] providesList = md1.getProvidesList();
+        for (InterfaceDescriptor prov : providesList) {
+          if (prov.isRegularHandler() && prov.isCompatible(req)) {
+            found = true;
           }
         }
-        if (!found) {
-          return false;
-        }
+      }
+      if (!found) {
+        return false;
       }
     }
     return true;
   }
 
-  static void topoSort(List<ModuleDescriptor> modules) {
-    List<ModuleDescriptor> result = new LinkedList<>();
+  /**
+   * Check if md's provided interfaces are required by any of modules.
+   * Only interfaces listed in allProvided are checked, this allows to exclude some optional
+   * interfaces from the check.
+   *
+   * @param modules     the modules that check against
```

Comment:
"The modules to check against." -- just add capital and period.

Reply:
I don't use capitals anywhere for parameters as first letter. So doing that now would be a mix of styles.

Reply:
Same story with ending period. The last sentence do not end in period.